python/src/aac/io/files/find.py

```diff
@@ -0,0 +1,42 @@
+"""Module to comb directories and identify AaC files."""
+import logging
+import os
+
+from aac.io.parser import parse, ParserError
+from aac.io.files.aac_file import AaCFile
+
+
+def find_aac_files(root_directory_to_search: str) -> list[AaCFile]:
+    """Return a list of paths to AaC files in or under the root directory."""
+    aac_files = []
+
+    root_dir_abs_path = os.path.abspath(root_directory_to_search)
+    if os.path.isdir(root_dir_abs_path):
+        directory_results = os.walk(root_dir_abs_path)
+        for directory_path, directory_names, filenames in directory_results:
+            for filename in filenames:
+                filepath = os.path.join(directory_path, filename)
+                if is_aac_file(filepath):
+                    aac_files.append(AaCFile(filepath, True, False))
+    else:
+        logging.error(f"Root path '{root_dir_abs_path}' for AaC file discovery is not a directory.")
+
+    return aac_files
+
+
+def is_aac_file(filepath: str) -> bool:
+    """Test if a target file is considered a valid AaC file."""
+    filename, file_ext = os.path.splitext(filepath)
+
+    is_valid_aac_file = False
+    if file_ext == ".yaml" or file_ext == ".aac":
+        try:
+            parse(filepath)
+        except ParserError as error:
+            logging.info(f"File '{filepath}' is not a valid AaC file. Reason: {error.errors}")
```

Comment:
ParserError.errors is a list of strings. We can write a story to standardize/modify these errors and their structures.

Reply:
I'd support a new story for standardizing that; however, my suggestion was more geared towards the output of this message since it will show a list of strings which can make the output harder to read. I think the solution would be to join all the error strings with newlines into a single string and output that.

Reply:
I think the log handles lists pretty well already. This is what the output looks like in the log:
```
07/18/2022 20:39:00::find::find.py::L37::INFO::File '/workspace/AaC/python/src/aac/main-branch.yaml' is not a valid AaC file. Reason: ["Failed to parse file: 'str' object has no attribute 'get'", '# This workflow is only intended to run when a pull-request is merged into the main branch\nname: Main branch Workflow\n\non:\n  push:\n    branches: [main]\n\njobs:\n  build_python_artifacts:\n    name: Python Build\n    uses: ./.github/workflows/python-build-artifacts.yml\n\n  build_vscode_artifacts:\n    name: VSCode Extension Build\n    needs: build_python_artifacts\n    uses: ./.github/workflows/vscode-build-artifacts.yml\n\n  python_tests:\n    name: Python Tests\n    needs: build_python_artifacts\n    uses: ./.github/workflows/python-tests.yml\n']
```

Reply:
I'd agree if this wasn't for logging. Multi-line comments do make programmatic parsing of logs ever so slightly more difficult.

Reply:
Per our discussion, we will aim to make logs easily machine-parseable so we can (eventually) create a log viewer that will alleviate some of the readability issues, I'm fine leaving this as-is.